### ISSUE TITLE
[MPS] reflection_pad1d backward returns wrong gradients when input length > 65536 and not a multiple of 65536

### ISSUE THREAD

**Issue by himidlow-software** [pytorch/pytorch#196949]
### 🐛 Describe the bug

On MPS, the **backward** of `F.pad(x, (p, p), mode='reflect')` on a 1-D signal (also reached through `torch.stft(..., center=True)`, whose default `pad_mode` is reflect) returns a wrong gradient whenever the input length along the padded dim is **greater than 65 536 and not a multiple of 65 536**. The forward is correct. The wrong entries begin at exactly index `T mod 65536` and run to the end of the signal, which looks like a grid/threadgroup sizing error in `reflection_pad1d_backward`. Pad size (1 … 2048) and batch size do not matter.

Measured against a float64 CPU reference (cosine similarity of the full gradient; a random per-position weighting so every position contributes):

| T | cos(mps fp32, cpu fp64) | wrong positions |
|---|---|---|
| 48 000 | 1.000000 | – |
| 65 535 | 1.000000 | – |
| 65 536 | 1.000000 | – |
| **65 537** | **0.0015** | 1 … 65 536 |
| 70 000 | 0.27 | 4 464 … 69 999 |
| 95 520 | 0.56 | 29 984 … 95 519 |
| 131 072 | 1.000000 | – |
| 200 000 | 0.15 | 3 392 … 199 999 |
| 384 000 | 0.39 | 56 320 … 383 999 |

The first wrong index is `T mod 65536` in every failing row. A slice-and-flip reflect pad (`torch.cat([x[...,1:p+1].flip(-1), x, x[...,-p-1:-1].flip(-1)], -1)`) is bit-identical in the forward and its backward matches the fp64 reference at every length, which is the workaround we are using.

This silently affects any STFT-domain loss trained on MPS on audio crops longer than 65 536 samples (1.37 s at 48 kHz) — the loss value looks normal, only the gradient is wrong. It went unnoticed in our training for several models.

Minimal reproduction:

```python
import torch
def cos(u, v): return float((u.flatten() @ v.flatten()) / (u.norm() * v.norm()))
torch.manual_seed(0)
for T in (65536, 65537, 95520, 131072, 200000):
    p = 1024
    x0 = torch.randn(1, T); w = torch.randn(1, T + 2 * p)
    g = []
    for dev, dt in (('cpu', torch.float64), ('mps', torch.float32)):
        a = x0.clone().to(dev, dt).requires_grad_(True)
        y = torch.nn.functional.pad(a, (p, p), mode='reflect')
        (y * w.to(dev, dt)).sum().backward()
        g.append(a.grad.detach().cpu().double())
    bad = ((g[1] - g[0]).abs() > 1e-3 * g[0].abs().max()).nonzero()[:, -1]
    print('T=%7d  cos=%.6f  first wrong index=%s' % (T, cos(g[1], g[0]), int(bad.min()) if len(bad) else '-'))
```

Output on the machine below:

```
T=  65536  cos=1.000000  first wrong index=-
T=  65537  cos=0.001522  first wrong index=1
T=  95520  cos=0.562817  first wrong index=29984
T= 131072  cos=1.000000  first wrong index=-
T= 200000  cos=0.147649  first wrong index=3392
```

Control ops at the same lengths (`unfold`, `rfft`, `Linear`, plain L1/L2, `torch.stft(center=False)`, `torch.stft(center=True, pad_mode='constant')`) all match the reference at cos 1.000000, so it is specific to the reflect-pad backward.

### Versions

- PyTorch 2.8.0 (pip wheel), torchaudio 2.8.0
- Python 3.9.6 (Apple Command Line Tools)
- macOS 26.5 (Darwin 25.5.0), Apple M4 Max, 128 GB
- `torch.backends.mps.is_available()` → True

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by Isalia20** [comment]
This seems like an MPS Graph issue since I can't reproduce this on macOS 27, looking into it and will put up a fix soon
